engine/lib/dependencies.ml

```diff
@@ -453,16 +457,33 @@ module Make (F : Features.T) = struct
                     ( name,
                       Concrete_ident.Create.move_under ~new_parent:new_name name
                     )))
+        | Some { v = Trait { items; _ }; _ }
+          when !includes_for_bundled_trait_methods ->
+            List.map items ~f:(fun { ti_ident; _ } ->
+                ( ti_ident,
+                  Concrete_ident.Create.move_under ~new_parent:new_name ti_ident
+                ))
         | _ -> []
       in
       let variant_and_constructors_renamings =
         List.concat_map ~f:variants_renamings renamings
         |> List.concat_map ~f:(fun (old_name, new_name) ->
-               [
-                 (old_name, new_name);
-                 ( Concrete_ident.Create.constructor old_name,
-                   Concrete_ident.Create.constructor new_name );
-               ])
+               let trait_methods_renamings =
+                 match from_ident old_name with
+                 | Some { v = Trait { items; _ }; _ }
+                   when not !includes_for_bundled_trait_methods ->
```

Comment:
I was expecting to see a condition based on `includes_for_bundled_trait_methods` only at the place where you create `Alias`. I don't get why there are two.
Unrelated to this PR, I think the arguments to `shallow_copy` are getting unclear. I wonder if we should not inline this function: the argument `variants_renamings` looks unrelated to the notion of "making a shallow copy".

Reply:
Yes, maybe we should refactor. The thing is where we create aliases, we also do the renaming. But in case we don't create aliases we still want to do the renaming (but we don't want to do it twice in the other case). So that's why we have this condition here (but maybe we could improve).